src/lib/error/api-error.test.ts

```diff
@@ -333,3 +338,79 @@ describe('Error serialization special cases', () => {
         expect(json.details!![0].validationErrors).toBe(results.errors);
     });
 });
+
+describe('Error serialization special cases', () => {
+    it('AuthenticationRequired: adds `path` and `type`', () => {
+        const type = 'password';
+        const path = '/api/login';
+        const error = new AuthenticationRequired({
+            type,
+            path,
+            message: 'this is a message',
+        });
+
+        const json = error.toJSON();
+
+        expect(json.path).toBe(path);
+        expect(json.type).toBe(type);
+    });
+
+    it('NoAccessError: adds `permission`', () => {
+        const permission = 'x';
+        const error = new NoAccessError(permission);
+        const json = error.toJSON();
+
+        expect(json.permission).toBe(permission);
+    });
+
+    it('BadDataError: adds `details` with error details', () => {
+        const description = 'You did **this** wrong';
+        const error = new BadDataError(description).toJSON();
+
+        expect(error.details![0].message).toBe(description);
+        expect(error.details![0].description).toBe(description);
+    });
+
+    it('OwaspValidationErrors: adds `validationErrors` to `details`', () => {
+        const results = owasp.test('123');
+        const error = new OwaspValidationError(results);
+        const json = error.toJSON();
+
+        expect(json.message).toBe(results.errors[0]);
```

Comment:
IMO it's much more readable when tests use the actual object shape in one expect assertions instead of 3 different expect calls. The key is to use https://jestjs.io/docs/expect#tomatchobjectobject

Reply:
I agree in most cases. I'll look aver the code and make any necessary adjustments.

Reply:
Yeah, I agree! Didn't realize you could put matchers inside `toMatchObject`, but TIL 😄  c21a299af6124d0a0dfb9c4ffd7ef13e28f66cba